File: anomaly-tracking-api/AnomalyTracking.Model/AnomalyDeclarations/AnomalyDeclaration.cs
using Shared.Core.Model.Common;
using Shared.Core.Model.Core;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace AnomalyTracking.Model.AnomalyDeclarations
{
    /// <summary>
    /// DTO that represents AnomalyDeclaration object.
    /// </summary>
    [DataContract]
    public class AnomalyDeclaration : EntityBase
    {
        /// <summary>
        /// AnomalyDeclaration's MoldId.
        /// </summary>
        [DataMember]
        public int UserId { get; set; }


        /// <summary>
        /// from tthe Cavity id we get the project and the mold
        /// </summary>
        [DataMember]
        public int CavityId { get; set; }


        /// <summary>
        /// here we stock the processId
        /// </summary>
        [DataMember]
        public int ProcessId { get; set; }


        /// <summary>
        /// from the anomalyId we get the face/position and anomalyType
        /// </summary>
        [DataMember]
        public int AnomalyId { get; set; }


      
        


       
      
    }
}

File: anomaly-tracking-webapp/src/app/views/anomalytracking.domain/_shared/models/anomalyDeclaration/anomalyDeclaration.ts
import {
  EntityBase,
  IEntityBaseInfo,
} from 'src/app/views/core.domain/shared/models/common/entitybase';
import { Cavity } from '../cavity/cavity';

/***
 * Represents the anomalyDeclaration object info.
 */
export interface IAnomalyDeclarationInfo extends IEntityBaseInfo {

  UserId: number;
  CavityId: number;
  ProcessId: number;
  AnomalyId?: number;
}

/**
 * @Class anomalyDeclaration
 * @classdesc Represents a anomalyDeclaration object.
 */
export class AnomalyDeclaration extends EntityBase {

  UserId: number;
  CavityId: number;
  ProcessId: number;
  AnomalyId?: number;

  /**
   * @constructor
   * Creates an instance of project.
   */
  constructor(info: IAnomalyDeclarationInfo) {
    super(info);
    this.UserId = info.UserId;
    this.CavityId = info.CavityId;
    this.ProcessId = info.ProcessId;
    this.AnomalyId = info.AnomalyId;
  }
}
